src/main/kotlin/uk/gov/dluhc/printapi/service/PrintRequestsFileProducer.kt

```diff
@@ -111,19 +111,30 @@ class PrintRequestsFileProducer {
                 eroDeliveryTownEn ?: "",
                 eroDeliveryAreaEn ?: "",
                 eroDeliveryPostcodeEn,
-                eroNameCy ?: "",
-                eroPhoneNumberCy ?: "",
-                eroEmailAddressCy ?: "",
-                eroWebsiteCy ?: "",
-                eroDeliveryStreetCy ?: "",
-                eroDeliveryPropertyCy ?: "",
-                eroDeliveryLocalityCy ?: "",
-                eroDeliveryTownCy ?: "",
-                eroDeliveryAreaCy ?: "",
-                eroDeliveryPostcodeCy ?: "",
+                getWelshValue(eroNameCy, eroNameEn),
+                getWelshValue(eroPhoneNumberCy, eroPhoneNumberEn),
+                getWelshValue(eroEmailAddressCy, eroEmailAddressEn),
+                getWelshValue(eroWebsiteCy, eroWebsiteEn),
+                getWelshValue(eroDeliveryStreetCy, eroDeliveryStreetEn),
+                getWelshValue(eroDeliveryPropertyCy, eroDeliveryPropertyEn),
+                getWelshValue(eroDeliveryLocalityCy, eroDeliveryLocalityEn),
+                getWelshValue(eroDeliveryTownCy, eroDeliveryTownEn),
+                getWelshValue(eroDeliveryAreaCy, eroDeliveryAreaEn),
+                getWelshValue(eroDeliveryPostcodeCy, eroDeliveryPostcodeEn)
             )
         }
 
+    private fun PrintRequest.getWelshValue(welshValue: String?, englishValue: String?): String {
+        val isWelshAndWelshPropertiesMissing =
+            certificateLanguage == PrintRequest.CertificateLanguage.CY && eroNameCy == null
```

Comment:
I checked if the `eroNameCy` is null or not to determine if the Welsh details are missing or not. The reason for that is, as can be seen in Ero English details, some properties can be null like `eroDeliveryTownEn`, `eroDeliveryLocalityEn`.

Assume we had this implementation instead, checking if Welsh properties are missing by checking the `welshValue` is null or not, instead of checking the `eroNameCy`.

<pre>
val isWelshAndWelshPropertiesMissing =
            certificateLanguage == PrintRequest.CertificateLanguage.CY && <b>welshValue</b> == null
</pre>

In that case, if we had an actual Welsh address where some optional fields like `eroDeliveryTownCy` and `eroDeliveryLocalityCy` are null and `eroDeliveryTownEn`  and `                eroDeliveryLocalityEn ` with non null values, than those English values would be mixed with the Welsh address.

I added the `should write file to stream given print request given full English properties and partially missing Welsh properties and certificate language is Welsh` test to specifically check that doesn't happen.